### PR TITLE
Performance telemetry for awesomebar suggestion provider durations

### PR DESCRIPTION
**Pull request by grigoryk**
See https://github.com/mozilla-mobile/android-components/pull/6802 for details; requires that PR.

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by grigoryk** [comment]
TODO: data review form & update PR w/ links.

File: app/src/main/java/org/mozilla/fenix/components/metrics/Metrics.kt [hunk 1]
```diff
@@ -474,6 +483,25 @@ private fun Fact.toEvent(): Event? = when (Pair(component, item)) {
...
+                is SessionSuggestionProvider -> PerfAwesomebar.sessionSuggestions
+                is SearchSuggestionProvider -> PerfAwesomebar.searchEgnineSuggestions
+                is ClipboardSuggestionProvider -> PerfAwesomebar.clipboardSuggestions
+                is ShortcutsSuggestionProvider -> PerfAwesomebar.shortcutsSuggestions
+                // NB: add PerfAwesomebar.syncedTabsSuggestions once we're using SyncedTabsSuggestionProvider
+                else -> {
+                    Logger("Metrics").error("Unknown suggestion provider: $provider")
+                    null
+                }
+            }?.setRawNanos(providerTiming.second as Long)
```
Comment (mcomella) [inline app/src/main/java/org/mozilla/fenix/components/metrics/Metrics.kt:501]: A `timespan` metric type can only be set once per ping: subsequent measurements will overwrite prior measurements. I think you'll need to use the `timing_distribution` metric type to record these values: https://mozilla.github.io/glean/book/user/metrics/timing_distribution.html

Comment (grigoryk) [reply app/src/main/java/org/mozilla/fenix/components/metrics/Metrics.kt]: Thanks for the catch!

**Comment by grigoryk** [comment]
# Request for data collection review form

**All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.**

1) What questions will you answer with this data?

This data will allow us to monitor real-life performance of all of the awesomebar suggestion providers. How fast/slow they are, what does distribution of duration looks like, etc.

2) Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements? Some example responses:

Awesomebar is one of the main points of interaction with the browser for our users. It needs to be fast and responsive. Measuring real-life performance will allow us to understand better how users are experiencing this part of the product. 

- Establish baselines and measure changes (across releases) in awesomebar performance.
- Track what effect product changes have on the performance of the awesomebar.

3) What alternative methods did you consider to answer these questions? Why were they not sufficient?

There are qualitative alternatives to understanding raw performance of suggestion providers (and of the awesomebar as a whole) - they mainly take form of monitoring user feedback from various channels. The quality of the data is hard to rely on, and data collection/processing effort is high.

Automatic collection of raw performance metrics is both much more accurate, and doesn't involve a lot of effort to monitor over time once dashboards are established.

4) Can current instrumentation answer these questions?

No.

5) List all proposed measurements and indicate the category of data collection for each measurement, using the [Firefox data collection categories](https://wiki.mozilla.org/Firefox/Data_Collection) found on the Mozilla wiki.   

**Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.**

<table>
  <tr>
    <td>Measurement Description</td>
    <td>Data Collection Category</td>
    <td>Tracking Bug #</td>
  </tr>
  <tr>
    <td>Duration distribution of history suggestion queries</td>
    <td>Category 1</td>
    <td>https://github.com/mozilla-mobile/android-components/issues/4992</td>
  </tr>
  <tr>
    <td>Duration distribution of bookmark suggestion queries</td>
    <td>Category 1</td>
    <td>https://github.com/mozilla-mobile/android-components/issues/4992</td>
  </tr>
  <tr>
    <td>Duration distribution of search engine suggestion queries</td>
    <td>Category 1</td>
    <td>https://github.com/mozilla-mobile/android-components/issues/4992</td>
  </tr>
  <tr>
    <td>Duration distribution of session (tabs) suggestion queries</td>
    <td>Category 1</td>
    <td>https://github.com/mozilla-mobile/android-components/issues/4992</td>
  </tr>
  <tr>
    <td>Duration distribution of synced tabs suggestion queries</td>
    <td>Category 1</td>
    <td>https://github.com/mozilla-mobile/android-components/issues/4992</td>
  </tr>
<tr>
    <td>Duration distribution of clipboard suggestion queries</td>
    <td>Category 1</td>
    <td>https://github.com/mozilla-mobile/android-components/issues/4992</td>
  </tr>
<tr>
    <td>Duration distribution of shortcuts suggestion queries</td>
    <td>Category 1</td>
    <td>https://github.com/mozilla-mobile/android-components/issues/4992</td>
  </tr>
</table>

6) How long will this data be collected?  Choose one of the following:

Data currently expires on 2020-09-15.

7) What populations will you measure?

All channels, all user population. 

8) If this data collection is default on, what is the opt-out mechanism for users?

Standard Glean SDK opt-out.

9) Please provide a general description of how you will analyze this data.

Through redash dashboards.

10) Where do you intend to share the results of your analysis?

Fenix, a-c and a-s teams.

11) Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection? If so:

N/A

**Comment by grigoryk** [comment]
A-C PR landed, so this one can land once it gets a data review; I'll update the metrics.yaml file w/ a link to a data review response before landing.

**Comment by grigoryk** [comment]
Ah, I also need to bump the pinned nightly a-c version once the new release is out. Will do that in the morning.

File: app/metrics.yaml [hunk 2]
```diff
@@ -2134,3 +2134,110 @@ startup.timeline:
...
+    send_in_pings:
+      - metrics
+    type: timing_distribution
+    time_unit: millisecond
+    description: >
+      Duration of a history awesomebar suggestion query.
+    bugs:
+      - https://github.com/mozilla-mobile/android-components/issues/4992
+    data_reviews:
+      - TODO
```
Comment (liuche) [inline app/metrics.yaml:2149]: TODO

**Comment by grigoryk** [comment]
Bumped A-C version, fixed failing glean test; should get a green CI run now.

File: app/metrics.yaml [hunk 3]
```diff
@@ -2179,3 +2179,110 @@ startup.timeline:
...
+      Duration of a bookmarks awesomebar suggestion query.
+    bugs:
+      - https://github.com/mozilla-mobile/android-components/issues/4992
+    data_reviews:
+      - https://github.com/mozilla-mobile/fenix/pull/10276#pullrequestreview-411101979
+    notification_emails:
+      - fenix-core@mozilla.com
+      - gkruglov@mozilla.com
+    expires: "2020-09-15"
+  search_egnine_suggestions:
```
Comment (ekager) [inline app/metrics.yaml:2214]: nit - i prefer "engine"

**Comment by mcarare** [comment]
@grigoryk Does this telemetry need an extension for the expiry date? It has already expired on `2020-11-15`

**Pull request merged**